course3/homework-string/my_string.h

```diff
@@ -1,6 +1,30 @@
 #pragma once
 
+#include <memory>
+
 class MyString
 {
-	// TODO
+public:
+
+	MyString();
+	MyString(const char* inchar);
+	MyString(const std::size_t size);
+	MyString(const MyString&);
+	
+	std::size_t size() const;
+	bool empty() const;
+	const char* c_str() const;
+	
+	bool operator==(const MyString& other) const;
+	bool operator!=(const MyString& other) const;
+	char& operator[](const std::size_t i) const;
```

Comment:
This function is not const-correct. It is 'const', so it promises the uses that it will ot modify the string, yet it returns non-const reference to the strings content, which can be used to modify individual characters.

There should be two version of the operator, const and non-const.